src/main/java/io/asyncer/r2dbc/mysql/client/SslBridgeHandler.java

```diff
@@ -220,8 +220,10 @@ static MySqlSslContextSpec forClient(MySqlSslConfiguration ssl, ConnectionContex
                 .applicationProtocolConfig(null);
             String[] tlsProtocols = ssl.getTlsVersion();
 
-            if (tlsProtocols.length > 0) {
-                builder.protocols(tlsProtocols);
+            if (tlsProtocols.length > 0 || ssl.getSslMode() == SslMode.TUNNEL) {
```

Comment:
Suggestion:

```java
if (tlsProtocols.length > 0) {
    if (ssl.getSslMode() == SslMode.TUNNEL) {
        builder.protocols(tlsProtocols);
    }
} else if ...
```

Reply:
Hmm, the current code means that if `SslMode` is `TUNNEL`, then don't set `builder.protocols`?

Reply:
I studied on it. and my aim was to prevent the execution from proceeding to any further `else if` or `else` clause when `ssl.getSslMode() == SslMode.TUNNEL`. as additional `else if` or `else` clauses are necessary only when `ssl.getSslMode() != SslMode.TUNNEL`.

Reply:
>Hmm, the current code means that if SslMode is TUNNEL, then don't set builder.protocols if user does not specified it
> So if it is changed to and logic, it means that if SslMode is TUNNEL, then builder.protocols is set if the user specified it. This obviously changes its behavior

Yes, I noticed that the subsequent `else if (isTls13Enabled(context))` and `else` clauses are designed to set protocols based on the version of MySQL/MariaDB. However, these settings do not relate to `SslMode.TUNNEL`.

Reply:
Yep, `OLD_TLS_PROTOCOLS` is the default for the MySQL server, not the tunnel/proxy server default.

Reply:
Make sense to me.

Reply:
> BTW, @jchrys do you try make TUNNEL not use OLD_TLS_PROTOCOLS?

Yes, I've made efforts to ensure that `SslMode.TUNNEL` is not associated with `OLD_TLS_PROTOCOLS`.

Reply:
It looks like we're in agreement. May I close this conversation?